src/modules/hip/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,2065 @@
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 4][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 36);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 4][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 36);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 5][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 45);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 5][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 45);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 5][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 45);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 6][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 54);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 6][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 54);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 6][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 54);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 7][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 63);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 7][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 63);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 7][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 63);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 8][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 72);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 8][hipThreadIdx_x8], &sum_f24.f8[1], &filter_f81, 72);
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 8][hipThreadIdx_x8], &sum_f24.f8[2], &filter_f81, 72);
+        rpp_hip_adjust_range(dstPtr, &sum_f24);
+        rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &sum_f24);
+    }
+}
+
+__device__ float gaussian(int x, int y, float mulFactor)
+{
+    float expFactor = - ((x * x) + (y * y)) * mulFactor;
+    expFactor = expf(expFactor);
+    float res  = (expFactor * mulFactor) / PI;
+    return res;
+}
+
+__global__ void set_filter_values(float *filterTensor, float *stdDevTensor, int kernelSize, int N, int batchSize)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x);
+    if(id_x >= batchSize)
+        return;
+
+    float *filter = &filterTensor[id_x * N];
+    float stdDev = stdDevTensor[id_x];
+    int cnt = 0;
+    float mulFactor = 1 / (2 * stdDev * stdDev);
+    float kernelSum = 0.0f;
+    for(int i = 0; i < kernelSize; i++)
+    {
+        for(int j = 0; j < kernelSize; j++)
+        {
+            filter[cnt] = gaussian(i, j, mulFactor);
+            kernelSum += filter[cnt];
+            cnt++;
+        }
+    }
+    kernelSum = (kernelSum == 0.0f) ? 1.0f : (1.0f / kernelSum);
```

Comment:
Please think about if there is any possibility of kernelSum==0? Sum can only be 0 if all kernel values are 0. Each value = expFactor * mulFactor. If stdDev is pre-verified properly with checks mulFactor can never be 0 and I presume neither can expFactor. Please verify and remove this check.

Reply:
Done